### ISSUE TITLE
Ungraceful failure creating tree when you have "add plot" but not "add tree" permission

### ISSUE THREAD

**Issue by RickMohr** [OpenTreeMap/otm-core#2852]
1. Configure a role with "Add planting site" permission but not "Add tree" permission
2. Using that role, try to create a tree. You're allowed to go all the way and click "Done", but then we fail with "Failed to add feature" toast.

What would be more graceful?
* Cadillac -- change UI to "Add an empty planting site", and offer no tree-specific fields
* Cheaper -- skip (or grey out) step 2 if user lacks "Add tree" permission

As we consider this we could also think about the requested feature of an explicit UI path to adding an empty planting site (vs. now where you just leave out the tree data).

**Comment by RobinIsTheBird** [comment]
Taking the cheaper solution. The Cadillac solution is blocked on a spec and prioritization for the "Add an empty planting site" enhancement.

**Comment by RobinIsTheBird** [comment]
Do we have a design for how this option should appear in the current UI design, or do we need a little of @lederer 's time?

**Comment by dboyer** [comment]
I believe this was only whiteboarded and we don't have a wireframe. I'll get that.

**Comment by lederer** [comment]
### Sidebar steps

When adding a planting site, I propose:
- Replace all "tree" strings with "planting site" throughout all 3 steps, obviously. Note: Not "empty planting site", but simply "planting site". Brevity counts.
- Retain the second step, but its only fields should be `planting site width` and `planting site length`.
- In step 3, the "After I add this planting site…" options should be:
  - "Add another planting site"
  - "Continue editing this planting site"
  - "I'm done"

### "Add" controls

Since the top nav add control will be deprecated in Spruce Up, I recommend we not get too fancy with changing it. I think the proposal below should be easily implementable.

![add menus](https://cloud.githubusercontent.com/assets/128699/22798014/66d7b37c-eece-11e6-873e-476df0eae4ed.png)


Holler if I'm missing or misunderstanding anything.

**Comment by jwalgran** [comment]
@lederer Thanks for these mockups. The development team discussed them yesterday and did have one question.

Throughout the application, instead of hiding options, we have preferred to disable them, and show a tooltip explaining why they are disabled. Here is and example of an Add Tree button at the right of the map page

![screen shot 2017-02-10 at 9 26 27 am](https://cloud.githubusercontent.com/assets/17363/22834669/217f4394-ef73-11e6-9914-ed31ad67fbdc.png)

We wanted to ask you if your design purposefully breaks this convention. What do you think about always showing all the options and disabling ones that are not available to the current user.

**Comment by lederer** [comment]
Interesting. I hadn't picked up on that, and so didn't factor it into my design. Though I'd argue we don't need to fully adhere to either approach (hiding vs disabling). Consistency is a means, not an end. 

Consider Google Docs. In View-only mode, it **hides** the **menus** that don't apply at all to View mode. But *within* the remaining menus, any inactionable **menu items** are **disabled** rather than hidden. It's not the only solution to that problem, but it works.

In the case of Add a tree, I think it would be misleading, even frustrating, to have a big button that won't let you do what it says.

Put another way, we should prioritize the "add a planting site" use case (by giving it a big button) over the "convey transparency into overall system capabilities" use case (by showing an Add tree button) or the "demarcate the limits of your permissions" use case (by disabling said button).

Put yet another way, if the thing you *can't* do is given a big button, but the thing you *can* do is hidden behind a small menu button, I'd argue we're doing something wrong.

**Comment by RobinIsTheBird** [comment]
@lederer has a good point.

However, removing things the user can't do has other implications.

If the visitor isn't logged in, they can't add anything. Presently, in that case, clicking Add a <anything> redirects to the login page. But it's unclear what to show them on the map page. Back to everything, with everything enabled? Take it away entirely and let the visitor figure out that they need to login?

**Comment by jwalgran** [comment]
If we don't already, I would like to assume that anonymous users have the default role when making decisions about what things to show/enable. That removes the need to treat logged out as a special case.

**Comment by lederer** [comment]
@jwalgran Hmm. Well, we already alter the UI based on the user's permissions. (Indeed entire swaths of the admin UI are dedicated to fine-grained configuration of those permissions.) I would think the UI for anonymous users would be commensurately affected by their own (lack of) permissions. Not a special case, per se, but merely… a case, with its own requirements like any other.

@RobinIsTheBird I would prefer we not show any Add buttons for anonymous users and instead have a strong `Sign in` button. I think the whole world knows by now you have to sign into a system to edit it. Despite this, if we were to retain the Add buttons for anonymous users, we could give them a generic "Add to map" label while logged out.

**Comment by RobinIsTheBird** [comment]
@jwalgran, yes, the anonymous user presently defaults to the default role. But I wanted to hear @lederer's point of view. Thank you, @lederer.

**Comment by dboyer** [comment]
My preference is to display an Add to Map label to anonymous users that would then prompt them to login rather than switching it to a Sign In button. Sign In implies that they should already have an account. Add to Map is demonstrating that there is an action they can take on the map page, even if it does first require them to sign in.

**Comment by lederer** [comment]
Doh. I meant just ensure the existing Sign in button is sufficiently prominent. Didn't mean replace the Add button with Sign in. Sorry, should've been more clear.
